src/main/java/games/strategy/engine/message/RemoteInterfaceHelper.java

```diff
@@ -3,49 +3,25 @@
 import java.lang.reflect.Method;
 import java.util.Arrays;
 import java.util.Comparator;
-import java.util.logging.Level;
-import java.util.logging.Logger;
+import java.util.stream.IntStream;
 
 import games.strategy.util.Tuple;
 
 class RemoteInterfaceHelper {
-  private static final Logger logger = Logger.getLogger(RemoteInterfaceHelper.class.getName());
-
   static int getNumber(final String methodName, final Class<?>[] argTypes, final Class<?> remoteInterface) {
     final Method[] methods = remoteInterface.getMethods();
     Arrays.sort(methods, methodComparator);
-    if (logger.isLoggable(Level.FINEST)) {
-      logger.fine("Sorted methods:" + Arrays.asList(methods));
-    }
-    for (int i = 0; i < methods.length; i++) {
-      if (methods[i].getName().equals(methodName)) {
-        final Class<?>[] types = methods[i].getParameterTypes();
-        // both null
-        if (types == argTypes) {
-          return i;
-        } else if (types != null && argTypes != null && types.length == argTypes.length) {
-          boolean match = true;
-          for (int j = 0; j < argTypes.length; j++) {
-            if (!argTypes[j].equals(types[j])) {
-              match = false;
-              break;
-            }
-          }
-          if (match) {
-            return i;
-          }
-        }
-      }
-    }
-    throw new IllegalStateException("Method not found");
+
+    return IntStream.range(0, methods.length)
+        .filter(i -> methods[i].getName().equals(methodName))
+        .filter(i -> Arrays.equals(argTypes, methods[i].getParameterTypes()))
+        .findFirst()
```

Comment:
You might want to switch to findAny instead of findFirst in case we want to use parallel streams at some point, since the order should really not matter as there shouldn't be more than one element in this stream anyways

Reply:
no objection  here, updated:   d668db9